### PR TITLE
Clone for Config and other improvements

### PR DESCRIPTION
**Pull request by antiguru**
* Clone for timely and communication config. Forces the cluster log function to be wrapped in an Arc, which is what happens anyways, so a no-op change.
* Use `execute_from` in `execute` to deduplicate code.
* Remove weird logging code from `execute`.

### CODE REVIEW

File: communication/src/allocator/zero_copy/initialize.rs [hunk 1]
```diff
@@ -39,7 +39,8 @@ pub fn initialize_networking(
     my_index: usize,
     threads: usize,
     noisy: bool,
-    log_sender: Box<dyn Fn(CommunicationSetup)->Option<Logger<CommunicationEvent, CommunicationSetup>>+Send+Sync>)
+    log_sender: Arc<Box<dyn Fn(CommunicationSetup)->Option<Logger<CommunicationEvent, CommunicationSetup>>+Send+Sync>>,
```
Comment (frankmcsherry) [inline communication/src/allocator/zero_copy/initialize.rs:42]: I think you can skip the `Box`, as `Arc<dyn Trait>` is a thing!

File: timely/src/execute.rs [hunk 2]
```diff
@@ -218,6 +219,17 @@ where
 /// // the extracted data should have data (0..10) thrice at timestamp 0.
 /// assert_eq!(recv.extract()[0].1, (0..30).map(|x| x / 3).collect::<Vec<_>>());
 /// ```
+///
+/// ### Communication logging
+///
+/// For multi-process (cluster) configurations, this functions installs a custom log
+/// function, replacing any previous function. The function connects to the host in
+/// the environment symbol `TIMELY_COMM_LOG_ADDR`.
```
Comment (frankmcsherry) [inline timely/src/execute.rs:227]: Should we just delete this feature?

**Review by frankmcsherry — APPROVED** [review]

File: communication/src/allocator/zero_copy/initialize.rs (hunk 1 above)
Comment (antiguru) [reply communication/src/allocator/zero_copy/initialize.rs]: Nice!

File: timely/src/execute.rs (hunk 2 above)
Comment (antiguru) [reply timely/src/execute.rs]: Tossed!

**Pull request merged**